test/unit/FtlControlConnectionUnitTests.cpp

```diff
@@ -86,6 +87,28 @@ class FtlControlConnectionUnitTestsFixture
         return controlConnections.at(connection);
     }
 
+    /**
+     * @brief Waits for the condition to return true, or returns false if timeout is hit
+     */
+    bool WaitFor(
+        std::function<bool()> condition,
+        std::chrono::milliseconds timeout = std::chrono::milliseconds{100})
+    {
+        std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+
+        while(!condition())
+        {
+            if (std::chrono::steady_clock::now() - begin > timeout)
+            {
+                INFO("Timed out waiting for condition");
+                return false;
+            }
+            std::this_thread::sleep_for(std::chrono::microseconds(1));
```

Comment:
Maybe we can open an issue to track adding more friendly async helpers to the fixture

Reply:
That'd be sweet, this feels a bit hacky but it is reliable.